s12/protobuf/proto/validator_helpers.go

```diff
@@ -17,12 +17,14 @@ const (
 const (
 	uuid     string = "^[0-9a-f]{8}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{12}$"
 	legacyId string = "(?i)^[0-9a-f]{8}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{12}(-?[0-9a-f]{2,5}-?[0-9a-f]{16})?$"
+	s12Id    string = "^\\w*_([0-9a-f]){32}$"
```

Comment:
The prefix regex part could be more specific. I believe only `a-z` are allowed here, so we don't need other characters that are in the `\w` class like digits and `_`.